### PR TITLE
Feature/add searchbar labels table route

### PR DESCRIPTION
**Pull request by efren02**
GitHub Link: https://github.com/framgia/sph-els-efren/tree/feature/add_searchbar_labels_table_route
Asana: https://app.asana.com/0/1202711011113639/1202711102078268/f

**Commands**
```
$ git config --global user.name "Bob"
$ git config --global user.email "bob@example.com"
```

**Definition of done**
- Added Search bar and Labels Component 
- Created a GitHub labels API component
- Created a labels table and put respective colors on labels
- Created a route for labels & issues for navigation 
-I've also managed to change my personal user  account into companies user account

**Notes**
- Need to refactor the GitHub labels into one component and make it dynamic for pagination. my goal right now is to implement all the functions and refactor them later.

**Screenshots**
image
image
image
image

### CODE REVIEW

File: src/api/github.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import axios from 'axios';
 
-export default axios.create({
+export default  axios.create({
```
Comment (chevyB) [inline src/api/github.js]:
```suggestion
export default axios.create({
```

File: src/api/github_labels.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import axios from 'axios';
+//this is only temporary need to refactor into 1 component but function is working
+export default  axios.create({
+    baseURL: 'https://api.github.com/repos/vuejs/vue/labels',
+   
+  })
```
Comment (chevyB) [inline src/api/github_labels.js]: Create a axios base so you dont create axios instance again and again. Follow DRY process.

Your .env
```
REACT_APP_GITHUB_API=https://api.github.com/repos/vuejs/vue
```

```suggestion
export default  axios.create({
  baseURL: `${process.env.REACT_APP_GITHUB_API}`,
})
```

then when fetching, you can reuse your axios base like this.
```
await API.get('/labels');
await API.get('/issues');
```

File: src/components/App.js [hunk 3]
```diff
@@ -19,19 +22,59 @@ import '../style.css';
...
             setIssue(response.data)
         }
-        
-    loadIssues();
-    //add pagination feature in content component
+
+        const loadLabel = async () => {
+            const response = await git_label.get();
+            setLabel(response.data)
+        }
```
Comment (chevyB) [inline src/components/App.js]:
```suggestion
      const loadLabel = async () => {
          const response = await git_label.get();
          setLabel(response.data)
      }
```
Proper indentions

File: src/components/Label.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+const Label = ({labels}) => {
+     //for refactoring
+     let count=0;
+     count = labels.length;
```
Comment (chevyB) [inline src/components/Label.js]:
```suggestion
   //for refactoring
   let count=0;
   count = labels.length;
```

File: src/components/Label.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        ); 
+    })
+
+    return( 
+        <table className="ui table inverted">
+            <thead>
+                <tr>
+                    <th colSpan="3">
+                    {count} Labels
+                    </th>
```
Comment (chevyB) [inline src/components/Label.js]:
```suggestion
                    <th colSpan="3">
                      {count} Labels
                    </th>
```

File: src/components/Label.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            </thead>
+            <tbody>
+              {renderedList}
+            </tbody>
+        </table>
+    );
+
+}
+
+export default Label;
```
Comment (chevyB) [inline src/components/Label.js]: add new line

**Review by chevyB — CHANGES_REQUESTED** [review]

File: src/components/App.js [hunk 7]
```diff
@@ -16,25 +19,100 @@ import '../style.css';
...
+                gitState = '&state=all'
+            } else {
+                gitState = '&state=' + githubState.value
+            }
 
-        const loadIssues = async () => {
-            const response = await github.get();
+        if (pageNumber === 0) {
+            setpageNumber(1)
+            const response = await github.get('issues?page=' + pageNumber +gitState);
```
Comment (chevyB) [inline src/components/App.js]:
```suggestion
            const response = await github.get('/issues?page=' + pageNumber +gitState);
```
is it not `/issues...`? is it working with your current implementation? did you test it?

File: src/components/App.js [hunk 8]
```diff
@@ -16,25 +19,100 @@ import '../style.css';
...
+    },[])
+
         return (
             <div className='ui container'>
                 <Header />
+                <div className="ui grid" id="search-label-content">
+                    <div className="ten wide column" id="search-label-content-column">
+                        <div className="ui left action left icon input fluid">
+                            <DropDown selected={githubState} options={options} onSelectedChange={setgithubState} />
+                            <i className="search icon " id="icon-search" style={{ marginLeft:githubState.size }}></i>
```
Comment (chevyB) [inline src/components/App.js]:
```suggestion
                              <i className="search icon " id="icon-search" style={{ marginLeft:githubState.size }} />
```
indention and use single tag

File: src/components/App.js [hunk 9]
```diff
@@ -16,25 +19,100 @@ import '../style.css';
...
+                            <button className='ui basic grey button' onClick={() => setpageNumber(pageNumber -1)}>Prev</button>
+                            <button className='ui basic grey button' onClick={() => setpageNumber(pageNumber +1)}>Next</button>
+                    <Route path='/'>
+                            <Content issues={issue} />
+                    </Route>
+                    <Route path='/issues'>
+                            <Content issues={issue} />
+                    </Route>
+                    <Route path='/labels'>
+                           <Label labels={label}></Label>
```
Comment (chevyB) [inline src/components/App.js]:
```suggestion
                           <Label labels={label} />
```

File: src/components/DropDown.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    document.body.addEventListener("click",onBodyClick, {capture: true
+    });
+
+    return () => {
+        document.body.removeEventListener("click", onBodyClick, {
+            capture: true,
+        });
+    }
+
+}, []);
```
Comment (chevyB) [inline src/components/DropDown.js]: Add indention

File: src/components/App.js (hunk 7 above)
Comment (efren02) [reply src/components/App.js]: yes sir it's working sir.

File: src/components/DropDown.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+        return(
+            <div key={option.value} className="item" onClick={() => onSelectedChange(option)}>
+                {option.label}
+            </div>
+        );
+    });
+
+    return (
+            <div ref={ref} className="ui form">
```
Comment (chevyB) [inline src/components/DropDown.js]:
```suggestion
            <div onClick={() => setOpen(false)} className="ui form">
```
You can remove useRef and use this instead, much clean and easy

File: src/components/App.js [hunk 12]
```diff
@@ -16,25 +19,100 @@ import '../style.css';
...
+            setIssue(response.data)
+        } else {
+            const response = await github.get('/issues?page=' + pageNumber +gitState);
+            if(response.data.length  === 0){
+                setpageNumber(pageNumber -1)
+                const response = await github.get('/issues?page=' + pageNumber +gitState);
+                setIssue(response.data)
+            }
             setIssue(response.data)
         }
```
Comment (chevyB) [inline src/components/App.js]:
```suggestion
    const response = await github.get('/issues', {
      params: {
        page:  pageNumber ?? 1,
        state: githubState === 'all' ? 'all' : githubState.value
      }
    });
```
All these lines can be simplified by this

File: src/components/DropDown.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    return (
+            <div onClick={() => setOpen(!open)} className="ui form">
+                <div onClick={() => setOpen(!open)} className={`ui basic inverted floating dropdown button ${open ? 'visible active': ''}`}>
+                    <div className="text">{selected.label}</div>
+                    <div className={`menu ${open ? 'visible transition' : ''}`}>
+                        {renderedOptions}
+                    </div>
+                    <i className="dropdown icon" />
+                </div>
+        </div>
```
Comment (chevyB) [inline src/components/DropDown.js]: Proper indention

File: src/components/App.js (hunk 12 above)
Comment (efren02) [reply src/components/App.js]: thank you kaayo sir. learned so much here.

**Review by chevyB — APPROVED** [review]

**Pull request merged**